addons/mrp/wizard/mrp_product_produce.py

```diff
@@ -104,3 +105,12 @@ def do_produce(self, cr, uid, ids, context=None):
         self.pool.get('mrp.production').action_produce(cr, uid, production_id,
                             data.product_qty, data.mode, data, context=context)
         return {}
+
+    def default_get(self, cr, uid, fields, context=None):
+        if context is None:
+            context = {}
+        res = super(mrp_product_produce, self).default_get(cr, uid, fields, context=context)
+        prodcution = self.pool.get('mrp.production').browse(cr, uid, context.get('active_id'), context=context)
```

Comment:
Simplify code: remove context assignation, check active_model, don't see with you update default_get instead of a default value on product_uom.